extensions/cornerstone/src/init.js

```diff
@@ -1,40 +1,239 @@
 import React from 'react';
 import OHIF from '@ohif/core';
-import { Dialog, Input } from '@ohif/ui';
-import cornerstone from 'cornerstone-core';
+import { Input, Dialog, ContextMenuMeasurements } from '@ohif/ui';
+import cs from 'cornerstone-core';
 import csTools from 'cornerstone-tools';
 import merge from 'lodash.merge';
 import initCornerstoneTools from './initCornerstoneTools.js';
-import cornerstoneTools from 'cornerstone-tools';
-import initWADOImageLoader from './initWADOImageLoader.js';
+import './initWADOImageLoader.js';
+import getCornerstoneMeasurementById from './utils/getCornerstoneMeasurementById';
 import measurementServiceMappingsFactory from './utils/measurementServiceMappings/measurementServiceMappingsFactory';
-//
 import { setEnabledElement } from './state';
 
+// TODO -> Global "context menu open state", or lots of expensive searches on drag?
+
+let CONTEXT_MENU_OPEN = false;
+
+const { globalImageIdSpecificToolStateManager } = csTools;
+
+const TOOL_TYPES_WITH_CONTEXT_MENU = [
+  'Angle',
+  'ArrowAnnotate',
+  'Bidirectional',
+  'Length',
+  'FreehandMouse',
+  'EllipticalRoi',
+  'CircleRoi',
+  'RectangleRoi',
+];
+
+const _refreshViewport = () =>
+  cs.getEnabledElements().forEach(({ element }) => cs.updateImage(element));
+
 /**
  *
  * @param {Object} servicesManager
  * @param {Object} configuration
  * @param {Object|Array} configuration.csToolsConfig
  */
-export default function init({ servicesManager, configuration }) {
+export default function init({
+  servicesManager,
+  commandsManager,
+  configuration,
+}) {
   const {
     UIDialogService,
     MeasurementService,
     DisplaySetService,
   } = servicesManager.services;
 
-  const callInputDialog = (data, event, callback) => {
+  /* Measurement Service */
+  const measurementServiceSource = _connectToolsToMeasurementService(
+    MeasurementService,
+    DisplaySetService
+  );
+
+  const onRightClick = event => {
+    if (!UIDialogService) {
```

Comment:
Does / can / should this ever happen? Couldn't we bail out at the top of the parent function?